### PR TITLE
Apply homepage redesign

### PR DESCRIPTION
**Pull request by jacobdaddario**
This is a WIP PR. It's based off of #711, so the history is a bit muddy here since I wanted to point this PR at your repo. 

Additionally, I don't have any of the images necessary to populate the pages, so I'm just filling it in with gray right now.

- [x] My code contains tests covering the code I modified
- [x] I linted and tested the project with `bin/check`
- [ ] I added significant changes and product updates to the [changelog](CHANGELOG.md)

### CODE REVIEW

File: app/views/home/show.html+redesign.erb
Comment (jacobdaddario) [inline app/views/home/show.html+redesign.erb:174]: This is a super complex collage that I imagine we'd rather load as one asset instead of 12 or so different assets. If you'd prefer to not serve this as separate assets, I can mock that up.

**Comment by jacobdaddario** [comment]
@joemasilotti So this is ready for review, kinda. It's essentially in a finished state, but there's two things that should be noted before we merge that I need your help on.
1. There's no images in here currently, and I've just blocked everything out with gray squares. If you have the final image assets that you want implemented, I can drop those in whenever.
2. Something just isn't right with the redesign variants. If you pull this PR down right now, it'll probably look like a complete mess. That's because for some reason, I can't get the correct stylesheet to load whenever I put things into the different variants. If you revert the branch back to [2741de0](https://github.com/joemasilotti/railsdevs.com/pull/712/commits/2741de0fc7e534830a180f556228e8c8dcebf6b5), you'll find that everything looks great (if I do say so myself). I have absolutely no idea what's going on. I've picked at it for a while now, and both the original version and the redesign variant have messed up styles once you add in the variant switching.
Once we figure these two things out, this is ready to go out the door. If we get variants working, we could even ship this before we have the image assets ready. Just let me know on this, and we can handle it how you'd like.

File: app/views/home/show.html+redesign.erb
Comment (jacobdaddario) [inline app/views/home/show.html+redesign.erb:3]: I can't get the stylesheets to work with the variant setup correctly... More detail in the conversation thread.

**Comment by jacobdaddario** [comment]
@joemasilotti This is all rebased to be independent of #711 and the variants are now setup correctly.

This is ready to merge if you're willing to wait on adding in the images. Otherwise, I need the final image assets. Regardless, this and #711 seem to be good on my end now, so I'll move on to the other redesign two issues.

File: app/views/home/show.html+redesign.erb
Comment (joemasilotti) [reply app/views/home/show.html+redesign.erb:174]: Hmm, good question. I think for now we can render it as a single image. We can always come back to it and do the complex CSS layout later.

**Comment by joemasilotti** [comment]
Excellent! I'm going to hold off on reviewing this until #711 is merged in.

But until then, feel free to continue using gray areas for the images. I will fill those in myself after I merge this PR in.

File: app/assets/images/icons/solid/arrow_long_right.svg
Comment (joemasilotti) [inline app/assets/images/icons/solid/arrow_long_right.svg:3]: I think this is from Heroicons v2, right? I'd like to keep everything on [v1](https://v1.heroicons.com) for now and update them all at once.

File: app/views/shared/_footer.html.erb
Comment (joemasilotti) [inline app/views/shared/_footer.html.erb:6]: We don't need to ever set a value as `true` if we convert this to an array. For example:

```suggestion
          <%= link_to request.params.merge(locale: resolve_locale(locale)), class: ["text-base", "text-gray-500 hover:text-gray-900": !Feature.enabled?(:redesign), "text-typography hover:text-navy-mid": Feature.enabled?(:redesign)], "aria-label": t(".language_name_of_locale", locale: locale) do %>
```

Comment (joemasilotti) [reply app/views/shared/_footer.html.erb:6]: Also, there's already a ton of repetition in these links. And the conditional from the redesign is making them a bit hard to reason about.

Let's extract a partial or View Component to use for all of these. I think the only parameters need to be the title, the URL, and if it's an external/internal link.

File: config/locales/en.yml
Comment (joemasilotti) [inline config/locales/en.yml:458]: Let's nest these to make them a bit easier to understand.

```yaml
home:
  show:
    about_card_1:
      content:
      title:
    about_card_2:
      content:
      title:
    # etc.
```

File: config/locales/en.yml
Comment (joemasilotti) [inline config/locales/en.yml:474]:
```suggestion
      reviews_header: What others are saying
```

File: config/locales/en.yml
Comment (joemasilotti) [inline config/locales/en.yml:467]:
```suggestion
      explain_title: How it works
```

File: app/views/home/show.html+redesign.erb
Comment (joemasilotti) [inline app/views/home/show.html+redesign.erb:4]: Let's pull this out to a partial. (No VC because I don't think we will re-use it.)

Also, TIL about `bg-gradient-radial`. Nice!

File: app/views/home/show.html+redesign.erb
Comment (joemasilotti) [inline app/views/home/show.html+redesign.erb:67]: This feels like a great opportunity to pull out a `Marketing::Testimonial` View Component.

File: app/views/home/show.html+redesign.erb
Comment (joemasilotti) [inline app/views/home/show.html+redesign.erb:51]: Let's extract the name, title, and copy to a YAML file so we can easily make changes in the future. We are already using `config/testimonials.yml` for business testimonials. Let's move that to `config/testimonials/businesses.yml` and create a new `developers.yml` file for these.

File: app/views/home/show.html+redesign.erb
Comment (joemasilotti) [inline app/views/home/show.html+redesign.erb]: This could be a View Component, too. We are already reusing it below. Maybe `Marketing::Hero` with an optional subtitle/description?

File: app/views/home/show.html+redesign.erb
Comment (joemasilotti) [inline app/views/home/show.html+redesign.erb:107]: Let's make sure to move these to the translation file.

File: app/views/home/show.html+redesign.erb
Comment (joemasilotti) [inline app/views/home/show.html+redesign.erb:133]: Let's move this to a partial.

File: app/views/home/show.html+redesign.erb
Comment (joemasilotti) [inline app/views/home/show.html+redesign.erb:157]: These feel like another great View Component opportunity!

**Review by joemasilotti — CHANGES_REQUESTED** [review]
Looking good so far! The design is really coming along.

I merged in `main` and made a few changes. Don't worry about rebasing anymore – feel free to push commits to the top if this branch from now on.

I made a bunch of small requests throughout the code. But the design replication looks great. Most of my feedback is in reference to extracting View Components or partials, moving copy to translations or YAML files, and code formatting.

Let me know if any of this doesn't make sense or you have any questions. I'm going to start working on the image placeholders.

**Comment by joemasilotti** [comment]
The more I dive into the code the more I'm thinking that I want explicit mobile designs. There are too many opportunities along the responsive breakpoints to make an odd looking page.

**For now lets design 100% for desktop at 7xl+.** I'm going to bring the designer back in to sketch up mobile designs to make sure we do it right. Until then, I'm going to remove all the breakpoint-specific code so we don't get bogged down by it.

**Comment by jacobdaddario** [comment]
Okay! I'm going to make the rest of the requested changes today. I need to finish a few errands first.

File: app/assets/images/icons/solid/arrow_long_right.svg
Comment (jacobdaddario) [reply app/assets/images/icons/solid/arrow_long_right.svg:3]: Right on. I ran into this on a personal project and it ended up being too late to fix 😂 Definitely better to stay on top of it. Good catch

File: app/views/shared/_footer.html.erb
Comment (jacobdaddario) [reply app/views/shared/_footer.html.erb:6]: Noted on the array syntax. I had seen that used in Matt Swanson's article on the token helper, but I ran into issues using the syntax in another project when I tried. At the time I thought that it was weird seeing hash keys in an array anyways so I ended up using this approach. I wonder if I was using an old version of Ruby.

Do you know what's going on with this syntax to make it legal? Is the Ruby parser assuming that we're trying to embed a hash inside of the array?

File: config/locales/en.yml
Comment (jacobdaddario) [reply config/locales/en.yml:458]: TIL

File: config/locales/en.yml
Comment (jacobdaddario) [reply config/locales/en.yml:475]: @joemasilotti As a general rule, do you prefer sentence case?

File: app/views/shared/_footer.html.erb
Comment (joemasilotti) [reply app/views/shared/_footer.html.erb:6]: >  Is the Ruby parser assuming that we're trying to embed a hash inside of the array?

I believe that is what's going on. The array has two elements here, `"text-base"` and then a hash. I think it has something to do with how the last parameter to a method can be an "options hash". But I don't know for certain! Could be a good blog post.

File: config/locales/en.yml
Comment (joemasilotti) [reply config/locales/en.yml:475]: Yep!

**Comment by jacobdaddario** [comment]
I'm gonna finish this up tonight ideally.

File: config/testimonials/developers.yml
Comment (jacobdaddario) [inline config/testimonials/developers.yml:4]: Moving the testimonials into these config files made it _very_ hard to do the mid-sentence bolding that the designer put together. I thought of giving it a second key, but then you run into the challenge of figuring out where to render it in the VC template. There's some inelegant solutions we could pursue, but I wanted to get your thoughts on if it is worth it before doing that.

**Comment by jacobdaddario** [comment]
Okay, all comments have been addressed on this, and I've re-requested a review. Thanks for your extensive patience on this. Sorry it took a bit longer than I'd have hoped.

**Comment by joemasilotti** [comment]
Thanks @jacobdaddario! I'm going to slowly work through this over the next few days.

I think the majority of the work is done but I'm going to make some code style + personal tweaks. I'll @ you before merge if you want to give it one more once-over.

**Comment by jacobdaddario** [comment]
Yeah no worries. Is there any way that I can help reduce that work for you in the future? I'd hate to create a bunch of work for you.

**Comment by jacobdaddario** [comment]
Oh gosh, I forget this was here. Do you plan to review this still? If so, I'll go finish the other PR that's drafted.

Sorry for letting this slide.

**Comment by joemasilotti** [comment]
All good! Thanks for the reminder.

Sadly, I think that I'm going to close out the redesign PRs for now. They can't go live without the mobile views and those are taking longer than expected to get done.

I'm planning to do an audit of the designs and potentially simplify them a bit in the next few months. I'll make sure to reach out if you'd like to work on them again!

**Comment by jacobdaddario** [comment]
No worries! I was mostly worried that I left you hanging.

**Pull request closed without merging**
